utils/testutils/testutils.go

```diff
@@ -27,3 +31,37 @@ func ReadTestFile(name string) ([]byte, error) {
 		return data.Bytes(), nil
 	}
 }
+
+// GetInterface returns the best match of an interface that might be listening on a given port.
+// This is helpful when a test is being run in a CI environment under docker.
+func GetInterface(port int) string {
+	dial := func(iface string, port int) bool {
+		c, err := net.Dial("tcp", iface+":"+strconv.Itoa(port))
+		if err != nil {
+			return false
+		}
+		c.Close()
+		return true
+	}
+	// First, we check dockerhost
+	iface := "dockerhost"
+	if ok := dial(iface, port); ok {
+		return iface
+	}
+	// If not, we check localhost
+	iface = "localhost"
+	if ok := dial(iface, port); ok {
+		return iface
+	}
+	// If nothing works, we just attempt to use a hack and get the interface IP.
+	// https://stackoverflow.com/a/37212665/4962526.
+	if runtime.GOOS != "windows" {
+		cmd := exec.Command("bash", "-c", `ifconfig | grep -E "([0-9]{1,3}\.){3}[0-9]{1,3}" | grep -v 127.0.0.1 | awk '{ print $2 }' | cut -f2 -d: | head -n1`)
```

Comment:
`ifconfig` could not be available. I tend to use `ip address` at least in Linux (Ubuntu)

Reply:
See the SO link. `ip` is not there on OSX. Happy to change it to something else though.

Reply:
Take into account that `ifconfig` is deprecated. So maybe is good to make one for Linux and another for darwin